scripts/superdesk/upload/image-crop-directive.js

```diff
@@ -279,136 +278,132 @@
-                    container.empty();
-
-                    img = new Image();
-                    img.onload = function() {
-                        //prepend for z-index purposes.
-                        container.prepend(img);
-                        container.append(overlay);
-                        elem.append(container);
-                        drawPoint();
-                    };
-
-                    function updatePOI(offsetX, offsetY) {
-                        scope.$apply(function() {
-                            scope.point.x = Math.round(offsetX * 100 / img.width) / 100;
-                            scope.point.y = Math.round(offsetY * 100 / img.height) / 100;
-                            scope.onChange();
-                        });
+            }],
+            link: function(scope, element, attrs, vm) {
+                var circleRadius = 30 / 2;
+                var lineThickness = 2;
+                function drawPoint(img, poi) {
+                    if (!angular.isDefined(poi)) {
+                        poi = vm.poi;
                     }
-
-                    function leaveOverlay(e) {
-                        e.preventDefault();
-                        updatePOI(e.offsetX, e.offsetY);
-                        $rootScope.$broadcast('poiUpdate', scope.point);
-                        overlay.onmousemove = null;
-                        overlay.onmouseup = null;
-                        overlay.onmouseleave = null;
-                    }
-
-                    overlay.onmousedown = function(e) {
-                        e.preventDefault();
-                        updatePOI(e.offsetX, e.offsetY);
-                        $rootScope.$broadcast('poiUpdate', scope.point);
-
-                        overlay.onmousemove = function(e) {
-                            e.preventDefault();
-                            updatePOI(e.offsetX, e.offsetY);
-                        };
-
-                        overlay.onmouseup = leaveOverlay;
-                        overlay.onmouseleave = leaveOverlay;
-                    };
-
-                    img.src = src;
-
-                    container.append(pointElem);
-                    container.append(crossLeft);
-                    container.append(crossRight);
-                    container.append(crossTop);
-                    container.append(crossBottom);
-
-                }
-
-                function drawPoint() {
-                    pointElem.css({
-                        left: (scope.point.x * img.width) - 15,
-                        top: (scope.point.y * img.height) - 15
+                    var topOffset = (poi.y * img.height) - circleRadius;
+                    var leftOffset = (poi.x * img.width) - circleRadius;
+                    var verticalLeftOffset = leftOffset + circleRadius - (lineThickness / 2);
+                    var horizontalTopffset = topOffset + circleRadius - (lineThickness / 2);
+                    element.find('.image-point__poi').css({
```

Comment:
It's better to cache these objects in variables so the DOM doesn't have to be queried repeatedly.